tools/spec_coverage/lib/co19.dart

```diff
@@ -0,0 +1,68 @@
+import 'dart:io';
+
+class Co19 {
+  static const languageDirName = "Language";
+  late final TestDir root;
+  late final TestDir language;
+
+  Co19(String path) {
+    Directory dir = Directory(path);
+    if (!dir.existsSync()) {
+      throw Exception("Directory '$path' does not exists");
+    }
+    root = TestDir(dir);
+    for (TestDir td in root.subDirs) {
+      if (td.name == languageDirName) {
+        language = td;
+        break;
+      }
+    }
+  }
+}
+
+class TestDir {
+  final String path;
+  final List<TestDir> subDirs = [];
+  final List<TestFile> tests = [];
+
+  TestDir(Directory root) : path = resolvePath(root.path) {
+    List<FileSystemEntity> entities = root.listSync();
+    for (FileSystemEntity fse in entities) {
+      if (skip(fse)) {
+        continue;
+      }
+      FileStat fs = fse.statSync();
+      if (fs.type == FileSystemEntityType.directory) {
+        subDirs.add(TestDir(Directory(fse.path)));
+      } else if (fs.type == FileSystemEntityType.file) {
+        tests.add(TestFile(fse.path));
+      }
+    }
+  }
+
+  static String resolvePath(String relativePath) {
+    String basePath = Directory.current.path;
```

Comment:
So this utility _must_ be executed from the root of the package? OK, I think that's a rather common restriction (but it might be nice to have an error message if the current directory at startup doesn't look like a package root).